### ISSUE TITLE
Handle identical proteins or v. similar proteins

### ISSUE THREAD

**Issue by edraizen** [bouralab/Prop3D#9]
E.g with Ig, there may be multiple structures that are identical, but have different PDB codes. Currently if there a multiple bindings sites in one pdb, they are combined. Should they be combined for identical structures as well?

To handle this, I've run uclust on all sequences from the same superfamily (%id=0.95) and then combined all binding sites in the same/similar in structure to the representative (the structure with the highest resolution) if the alignment is a match in the same position. Does this approach sound reasonable? Should I increase/decrease the %Id? Choose a different representative?

**Comment by neksa** [comment]
What you call 'identical' structures may differ by one of a few residues (e.g. mutants) or have a different conformation or have a different interaction context.
For your application with deep learning minor differences in structure are insignificant so a highly conservative approach (80% sounds reasonable) could be the way to go. In my opinion the major threat is leakage between train and test sets - you don't want interfaces in the test set to be related closely to your training examples. In this regard, training superfamily-specific models is very tricky. I would even recommend not doing it, if you really want your network to generalize - train and test on unrelated superfamilies.

**Comment by edraizen** [comment]
Hi Alex, thanks for your response! 

I agree 80% is reasonable if I use superfamily-specific models. I originally wanted to use superfamily-specific models so I could account for the large amount of unlabeled binding sites with unsupervised pre-training. The data was also too large to train so I split it into superfamilies to make training faster.

However, I decided to move away from IBIS since the mssql database was too messy and took too much time to organize. I switched to use EPPIC binding sites, mapped to CATH domains. Now I can make use of the CATH cluster representatives for training.

> you don't want interfaces in the test set to be related closely to your training examples. 

The way I was planning on doing this was to map all binding sites in a CATH cluster to its cluster representative. Do you think a sequence OR structure alignment would be better to map binding sites from a cluster member to the cluster representative? One sample would then be the union of all binding site residues in the cluster mapped onto the cluster representative.

EPPIC does cluster interfaces at the whole chain level, but since I am looking at domains, their clustering is not as useful.

> In my opinion the major threat is leakage between train and test sets

If I don't use superfamly-specific models, I was planning on making sure all of a given superfamily are in one split of train or test. This is what AlphaFold did when training using CATH 35% cluster. For a superfamily-specific model I would just split the cluster representatives (at 60% or 90%?) randomly.

Thanks for your help! If you have any other suggestions, I'd love to know!

**Comment by neksa** [comment]
> 
> However, I decided to move away from IBIS since the mssql database was too messy and took too much time to organize. I switched to use EPPIC binding sites, mapped to CATH domains. Now I can make use of the CATH cluster representatives for training.
> 
EPPIC sounds good.

The fundamental difference to IBIS #1 is that IBIS is structure-based sequence alignment, while EPPIC is MSA; #2 is that IBIS is domain-based.

> you don't want interfaces in the test set to be related closely to your training examples.
> 
> The way I was planning on doing this was to map all binding sites in a CATH cluster to its cluster representative. Do you think a sequence OR structure alignment would be better to map binding sites from a cluster member to the cluster representative? One sample would then be the union of all binding site residues in the cluster mapped onto the cluster representative.
> 

I am against this idea because by learning from aggregated features (or choosing one representative) you would reduce the size of your dataset.
It is absolutely fine to have redundancy in Deep Learning examples. You only need to control that your training and testing examples are not related. Otherwise the net will learn the shortcut quickly.

> EPPIC does cluster interfaces at the whole chain level, but since I am looking at domains, their clustering is not as useful.
> 
Correct.

> In my opinion the major threat is leakage between train and test sets
> 
> If I don't use superfamly-specific models, I was planning on making sure all of a given superfamily are in one split of train or test.
> 
Sounds very good for learning generic properties of binding sites.

>  For a superfamily-specific model I would just split the cluster representatives (at 60% or 90%?) randomly.
> 
You may need a separation metric that is binding site based. How about Jaccard (overlap/union) of binding site residues? Or if you know the information about the binding partner you can make sure that the binding partner is from a different superfamily (it is actually the IBIS way).

Learning and testing on same binding site - just different homologs is a very easy task for a CNN - you need to push it :)

An important question I had: what is your ground zero model? You need to have a simpler reference model that you are comparing to. Sequence-based or structure based or something else…?

Can you share a recent presentation? I think it’s a great project, but you seem to be stuck - maybe I can “unstuck” you.

**Comment by edraizen** [comment]
> It is absolutely fine to have redundancy in Deep Learning examples. You only need to control that your training and testing examples are not related. Otherwise the net will learn the shortcut quickly.

I am just worried that if I do not annotate all known binding sites for a sample, the model will mark binding site residues from one sample as negatives even if they are binding sites, just not in this specific sample -- or even captured by xtallography/cryo-em/etc yet.

> You may need a separation metric that is binding site based. How about Jaccard (overlap/union) of binding site residues?
 
Thanks, that is an interesting idea to try. This still requires lots of structure-based alignments though?

> Or if you know the information about the binding partner you can make sure that the binding partner is from a different superfamily (it is actually the IBIS way).

This seems like a good way to go. The end goal of my model is to instead of predicting true/false if a residue is in binding site, it will predict a 6k binary vector if a residue binds a given superfamily

> An important question I had: what is your ground zero model? You need to have a simpler reference model that you are comparing to. Sequence-based or structure based or something else…?

Since it's not functional yet, I haven't compared it to other predictors. But when I do, it would be against structure-based models: either templates (IBIS or PRISM) or ml (RAD-T). Or do you mean implement my own simple svm?

> Can you share a recent presentation? I think it’s a great project, but you seem to be stuck - maybe I can “unstuck” you.

Haha yes, I am a bit stuck. Thank you so much for offering to help. Here is one of my recent talks I gave: https://docs.google.com/presentation/d/1nCUBwTRm_XKqY-QcwSG8RPIWCzh8lKzhnJSoYdFZP34/edit?usp=sharing
